cdap-spark-core-base/src/main/java/co/cask/cdap/app/runtime/spark/submit/AbstractSparkSubmitter.java

```diff
@@ -186,6 +186,7 @@ private void submit(SparkRuntimeContext runtimeContext, String[] args) {
     ImmutableList.Builder<String> builder = ImmutableList.builder();
 
     builder.add("--master").add(getMaster(configs));
+    builder.add("--deploy-mode").add("cluster");
```

Comment:
Better have it controlled by sub-class. Maybe just modify the `getMaster` to `setMaster(Map<String, String> configs, ImmutableList.Builder<String> argBuilder)`.